src/main/java/com/project/hireup/service/UserService.java

```diff
@@ -0,0 +1,101 @@
+package com.project.hireup.service;
+
+import static com.project.hireup.type.ErrorCode.ALREADY_AUTH;
+import static com.project.hireup.type.ErrorCode.EMAIL_NOT_SEND;
+import static com.project.hireup.type.ErrorCode.NOT_EQUAL_CONFIRM_PASSWORD;
+import static com.project.hireup.type.ErrorCode.NOT_EQUAL_TOKEN;
+import static com.project.hireup.type.ErrorCode.NOT_EXIST_EMAIL_AUTH_KEY;
+import static com.project.hireup.type.ErrorCode.USER_ALREADY_EXISTS;
+
+import com.project.hireup.component.MailComponent;
+import com.project.hireup.dto.SignUpRequestDto;
+import com.project.hireup.entity.User;
+import com.project.hireup.exception.HireUpException;
+import com.project.hireup.repository.UserRepository;
+import com.project.hireup.type.UserRole;
+import com.project.hireup.type.UserStatus;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+  private final UserRepository userRepository;
+  private final MailComponent mailComponent;
+
+  @Value("${admin.token}")
+  private String adminToken;
+
+  // 회원가입
+  public void signUp(SignUpRequestDto requestDto) {
+
+    // 회원가입 진행 -> 이미 등록된 email이 있다면 예외 처리
+    if (userRepository.existsByEmail(requestDto.getEmail())) {
+      throw new HireUpException(USER_ALREADY_EXISTS);
+    }
+
+    // 비밀번호와 비밀번호 확인 입력이 다를시 예외 처리
+    if (!requestDto.getPassword().equals(requestDto.getConfirmPassword())) {
+      throw new HireUpException(NOT_EQUAL_CONFIRM_PASSWORD);
+    }
+
+    System.out.println(requestDto.isAdmin());
+    // 기본으로는 일반 user
+    UserRole role = UserRole.ROLE_USER;
+    // isAdmin 값이 true 이고 함께 넣어준 토큰 값이 일치한다면 Partner role 부여
+    if (requestDto.isAdmin()) {
+      if (!adminToken.equals(requestDto.getAdminToken())) {
+        throw new HireUpException(NOT_EQUAL_TOKEN);
+      }
+      role = UserRole.ROLE_ADMIN;
+    }
```

Comment:
if문이 중첩으로 있으면 가독성이 좋지 않습니다. 또한 선언과 할당을 같이 해주어야 좋은 코드이며(모든 변수를 finalize 시킬수 있어야함) 재할당하는 부분은 가급적 피해야합니다
```suggestion
    UserRole role = requestDto.isAdmin() ? UserRole.ROLE_ADMIN : UserRole.ROLE_USER;
    
    if(role == UserRole.ROLE_ADMIN && !adminToken.equals(requestDto.getAdminToken())) {
        throw new HireUpException(NOT_EQUAL_TOKEN);
    }  
```

이 편이 좀더깔끔한것 같긴 한대요.. 최선인지는 잘 모르겠네요. 좀더 좋은 코드 방식이 있을 것 같긴하는데 지금으로서는 잘 생각이 안나네요. 어째뜬 가급적 재할당하는 부분은 최소화하고, if문은 가능하면 중첩되지 않고 풀어서 사용하는 것이 좋습니다

Reply:
@mingming-mentor 피드백 주신 부분 모두 수정하였습니다. 확인 부탁드립니다.